File: app/src/main/java/com/example/chuckjokes/Error/ErrorFragment.java
package com.example.chuckjokes.Error;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import com.example.chuckjokes.databinding.ErrorFragmentBinding;

import org.jetbrains.annotations.NotNull;

public class ErrorFragment extends Fragment {

    private final ErrorPresenter presenter;

    private ErrorFragmentBinding binding;

    public ErrorFragment(Exception err) {
        this.presenter = new ErrorPresenter(this, err);
    }

    public View onCreateView(@NotNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        binding = ErrorFragmentBinding.inflate(inflater);

        presenter.setError();

        binding.moreAboutErrorBut.setOnClickListener(unused -> presenter.onMoreBtnClicked());

        return binding.getRoot();
    }

    public void setBaseText(String text) {
        binding.errorBaseInfo.setText(text);
    }

    public int getExtendVisibility() {
        return binding.moreAboutError.getVisibility();
    }

    public void setExtendVisibility(int state) {
        binding.moreAboutError.setVisibility(state);
    }

    public void setExtendText(String text) {
        binding.moreAboutError.setText(text);
    }

}

File: app/src/main/java/com/example/chuckjokes/error/ErrorFragment.kt
package com.example.chuckjokes.error

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.chuckjokes.Shared
import com.example.chuckjokes.databinding.ErrorFragmentBinding
import com.example.chuckjokes.viewModelsExt
import java.lang.IllegalStateException

class ErrorFragment : Fragment() {

    private lateinit var binding: ErrorFragmentBinding

    private val viewModel by viewModelsExt {
        ErrorViewModel()
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        binding = ErrorFragmentBinding.inflate(inflater)

        binding.moreAboutErrorBut.setOnClickListener {
            viewModel.moreAboutErrorClicked()
        }

        viewModel.additionalInfoVisibility.observe(viewLifecycleOwner) {
            binding.moreAboutError.visibility = if (it) {
                View.VISIBLE
            } else {
                View.GONE
            }
        }

        viewModel.errorBaseInfoText.observe(viewLifecycleOwner) {
            binding.errorBaseInfo.text = it
        }

        viewModel.moreAboutErrorText.observe(viewLifecycleOwner) {
            binding.moreAboutError.text = it
        }

        viewModel.setArguments(
            try {
                requireArguments()
            } catch (ex: IllegalStateException) {
                /*
                    Это абсолютно рандомная мысль, но прикольно же!
                    это не должно переехать во вьюмодель?
                    это вообще нужно обрабатывать? Теоретически, это невозможно и бесполезно использовать без аргументов
                */
                val args = Bundle()
                args.putSerializable(Shared.ERROR_EXCEPTION_TAG, ex)
                args
            }
        )

        return binding.root
    }

    companion object ErrorFragmentFactory {
        fun newInstance(args: Bundle?): ErrorFragment {
            val myFragment = ErrorFragment()
            myFragment.arguments = args
            return myFragment
        }
    }
}
